framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2766,8 +2766,10 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
                         new ArrayList<>(superCon.getParameterTypes().size() + 1);
                 if (TreeUtils.hasSyntheticArgument(tree)) {
                     p.add(con.getParameterTypes().get(0));
+                    con.setReceiverType(superCon.getReceiverType());
                 } else if (con.getReceiverType() != null) {
                     p.add(con.getReceiverType());
+                    con.setReceiverType(superCon.getReceiverType());
```

Comment:
Can you explain the order of these? Why does it first add the old receiver of `con` to `p` and then change the receiver of `con`? Should these be switched?
For both these changes, can you add a comment what they do?

Reply:
Thanks! Yes, I need to switch the order.

Reply:
And added comments.